src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -288,8 +289,14 @@ await CreateAndUploadGameplaySession(
         private async Task MoveSegmentsToProject(string botSegmentsDirectoryPrefix)
         {
             // get all the file paths normalized to /
-            var segmentFiles = Directory.EnumerateFiles(botSegmentsDirectoryPrefix).Where(a=>a.EndsWith(".json")).Select(a=>a.Replace('\\','/')).Select(a=>a.Substring(a.LastIndexOf('/')+1));
-
+            var segmentFiles = Directory.EnumerateFiles(botSegmentsDirectoryPrefix)
+                .Where(a=>a.EndsWith(".json"))
+                .Select(a=>a.Replace('\\','/'))
+                .Select(a=>a.Substring(a.LastIndexOf('/')+1));
+
+            // Order numerically instead of alphanumerically to ensure 2.json is before 10.json.
+            segmentFiles = BotSegmentDirectoryParser.OrderJsonFiles(segmentFiles).ToList();
```

Comment:
calling .ToList() would make no difference in ordering and adds a layer of inefficiency, marking as requested changes until there is an explanation on why it wasn't working without this

Reply:
I just spent a good few minutes looking at this - to me it smelled like something to do with the fact that these LINQ functions defer execution until the elements are actually needed, but I couldn't find anything glaring in the code (for instance, we actually force the execution of the LINQ functions inside of OrderJsonFiles because of some print statements that print the elements coming in, so they are already being forced).

From the video, the strange part is that the logs just stop.... I would double check if an exception is being thrown somehow and if it's being suppressed by some surrounding try catch.

Reply:
Linq functions are also notorious for eating exceptions